Sources/OpenAPIRuntime/EventStreams/ServerSentEventsDecoding.swift

```diff
@@ -156,20 +180,26 @@ extension ServerSentEventsDeserializationSequence.Iterator {
         /// - Returns: An action to perform.
         mutating func next() -> NextAction {
             switch state {
-            case .accumulatingEvent(var event, var buffer):
+            case .accumulatingEvent(var event, var buffer, let predicate):
                 guard let line = buffer.first else { return .needsMore }
                 state = .mutating
                 buffer.removeFirst()
                 if line.isEmpty {
                     // Dispatch the accumulated event.
-                    state = .accumulatingEvent(.init(), buffer: buffer)
                     // If the last character of data is a newline, strip it.
                     if event.data?.hasSuffix("\n") ?? false { event.data?.removeLast() }
+                    
+                    state = .accumulatingEvent(.init(), buffer: buffer, predicate: predicate)
+                    
+                    if let data = event.data, !predicate(ArraySlice(data.utf8)) {
+                        state = .finished
+                        return .returnNil
+                    }
                     return .emitEvent(event)
```

Comment:
```suggestion
                    if let data = event.data, !predicate(ArraySlice(data.utf8)) {
                        state = .finished
                        return .returnNil
                    }
                    state = .accumulatingEvent(.init(), buffer: buffer, predicate: predicate)
                    return .emitEvent(event)
```

Minor - the state should only be assigned a single time in each pass.